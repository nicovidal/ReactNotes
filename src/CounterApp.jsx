import PropTypes from 'prop-types';



export const CounterApp = ({value}) => {
  return (
    <div>
    <div>CounterApp</div>
    <h2>{value}</h2>
    </div>
  )
}


/* export function CounterApp({value}){

    return(
        <div>
        <h1>CounterApp</h1>
        <h2>{ value }</h2>
        </div>
    )
} */

CounterApp.protoType={

    value:PropTypes.number.isRequired
}