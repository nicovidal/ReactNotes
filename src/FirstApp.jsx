
//son las propiedades a la funcion
//se debe destructurar
//un PropTypes sirven para definirle el tipo alas properties
import PropTypes from 'prop-types';


export const FirstApp =({title,subTitle,name})=>{
   title,
   subTitle
 

    return(
        <div>
        <h1 data-testid="test-title">{ title}</h1>
        {/* <code>{JSON.stringify(newMessage)}</code> */}   
        <p>{subTitle} </p>
        <p>{subTitle} </p>
        <p>{name}</p>
        </div>
    )
}


FirstApp.protoType={

    title:PropTypes.string.isRequired,
    subTitle:PropTypes.string.isRequired
}

//se puede definir cualquier cantidad de valores

FirstApp.defaultProps={
    /* title:'NO hay ningun Titulo', */
    subTitle:'No hay subtitulo',
    name:'nico',

}



