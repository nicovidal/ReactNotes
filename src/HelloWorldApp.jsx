/* export function App(){

    return(<h1>Hola Mundo!!</h1>)

} */

//tambien puede ser por defecto

//export default App;

//usando HelloWorldApp en el componente


export const HelloWorldApp = () => {
  return (
    <h1>HelloWorldApp</h1>
  );
}
