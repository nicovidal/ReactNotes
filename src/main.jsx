//elementos para renderizar la app
import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirstApp } from './FirstApp';
import { HelloWorldApp } from './HelloWorldApp';
import './styles.css';



ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <FirstApp title="Hola soy goku" subTitle={123}/> 
    </React.StrictMode>
)