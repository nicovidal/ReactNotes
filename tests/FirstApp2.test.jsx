import { render,screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";




describe('Pruebas en <first app/>', () => { 
  const subTitle='soy un subtitulo'
  const title='Hola,soy Nico'
  test('debe de hacer un match con el spapshot', () => { 
    const {container}=render(<FirstApp title={title}/>)
    expect(container).toMatchSnapshot();
   });

   test('debe de mostrar el mensaje "Hola,soy Nico"', () => { 

    render(<FirstApp title={title}/>);
    expect(screen.getByText(title)).toBeTruthy();
    /* screen.debug(); */



    });

    test('debe de mostar el titulo en un h1', () => { 

      render(<FirstApp title={ title }/>);
      expect(screen.getByRole('heading',{level: 1 }).innerHTML).toContain(title);

     });

     test('debe de mostrar el subitulo enviado por props', () => { 
        render(<FirstApp 
              title={ title }
              subTitle={ subTitle }
              />
              
            );
            expect(screen.getAllByText(subTitle).length).toBe(2);
      });
 });