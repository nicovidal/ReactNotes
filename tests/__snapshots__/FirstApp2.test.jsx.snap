// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <first app/> debe de hacer un match con el spapshot 1`] = `
<div>
  <div>
    <h1
      data-testid="test-title"
    >
      Hola,soy Nico
    </h1>
    <p>
      No hay subtitulo
       
    </p>
    <p>
      No hay subtitulo
       
    </p>
    <p>
      nico
    </p>
  </div>
</div>
`;
